src/app/qgshandlebadlayers.cpp

```diff
@@ -113,12 +119,13 @@ QgsHandleBadLayers::QgsHandleBadLayers( const QList<QDomNode> &layers )
 
     QString name = node.namedItem( QStringLiteral( "layername" ) ).toElement().text();
     QString type = node.toElement().attribute( QStringLiteral( "type" ) );
+    QString id = node.namedItem( QStringLiteral( "id" ) ).toElement().text();
     QString datasource = node.namedItem( QStringLiteral( "datasource" ) ).toElement().text();
     QString provider = node.namedItem( QStringLiteral( "provider" ) ).toElement().text();
     QString vectorProvider = type == QLatin1String( "vector" ) ? provider : tr( "none" );
     bool providerFileBased = ( QgsProviderRegistry::instance()->providerCapabilities( provider ) & QgsDataProvider::File ) != 0;
-    const QString basepath = datasource.left( datasource.lastIndexOf( '/' ) );
-    mFileBase[name].append( basepath );
+
+    mOriginalFileBase[id] = datasource.left( datasource.lastIndexOf( QDir::separator() ) );
```

Comment:
To understand the logic here -- this is retrieving the folder from a folder + file path, right?

If so, better to use the inbuilt Qt methods for this -- see the QFileInfo class.

Reply:
You're right, should be fixed

Reply:
true, should be ok now.